design-documents/graph-ql/coverage/negotiableQuotes.graphqls

```diff
@@ -0,0 +1,361 @@
+    grand_total: Float @doc(description: "Grand total")
+    base_grand_total: Float @doc(description: "Base grand total")
+    subtotal: Float @doc(description: "Sub total")
+    base_subtotal: Float @doc(description: "Base sub total")
+    discount_amount: Float @doc(description: "Discount amount")
+    subtotal_with_discount: Float @doc(description: "Subtotal with discount")
+    shipping_amount: Float @doc(description: "Shipping amount")
+    base_shipping_amount: Float @doc(description: "Base shipping amount")
+    shipping_discount_amount: Float @doc(description: "Shipping discount amount")
+    tax_amount: Float @doc(description: "Tax amount")
+    base_tax_amount: Float @doc(description: "base_tax_amount")
+    shipping_tax_amount: Float @doc(description: "shipping_tax_amount")
+    subtotal_incl_tax: Float @doc(description: "subtotal_incl_tax")
+    shipping_incl_tax: Float @doc(description: "shipping_incl_tax")
+    base_shipping_incl_tax: Float @doc(description: "base_shipping_incl_tax")
+    base_currency_code: String @doc(description: "base_currency_code")
+    quote_currency_code: String @doc(description: "quote_currency_code")
+    items_qty: Int @doc(description: "items_qty")
+    items: [ItemsOutput] @doc(description: "Items details")
+    total_segments: [TotalSegmentOutput] @doc(description: "Total segments")
+}
+
+type ItemsOutput {
+    item_id: ID @doc(description: "Item ID")
+    price: Float @doc(description: "Price")
+    base_price: Float @doc(description: "Base price")
+    qty: Int @doc(description: "Quantity")
+    row_total: Float @doc(description: "Row total")
+    base_row_total: Float @doc(description: "Base row total")
+    row_total_with_discount: Float @doc(description: "Row total with discount")
+    tax_amount: Float @doc(description: "Tax amount")
+    base_tax_amount: Float @doc(description: "Base tax amount")
+    tax_percent: Float @doc(description: "Tax percent")
+    discount_amount: Float @doc(description: "Discount amount")
+    base_discount_amount: Float @doc(description: "Base discount amount")
+    discount_percent: Float @doc(description: "Discount percent")
+    price_incl_tax: Float @doc(description: "Price incl. tax")
+    base_price_incl_tax: Float @doc(description: "Base price incl. tax")
+    row_total_incl_tax: Float @doc(description: "Row total incl. tax")
+    base_row_total_incl_tax: Float @doc(description: "Base row total incl. tax")
+    name: String @doc(description: "Name")
+}
+
+type TotalSegmentOutput {
+    code: String @doc(description: "Segment code")
+    title: String @doc(description: "Segment title")
+    value: Float @doc(description: "Segment value")
+}
+
+type TotalInfo {
+    totals: TotalsOutput! @doc(description: "List of order otals for negotiable quote")
+}
+
+input UpdatedPricesInput {
+    quote_ids: [String]! @doc(description: "IDs of the updated quotes")
+}
+
+input NegotiableQuoteRequestInput {
+    quote_id: String! @doc(description: "Quote hash")
```

Comment:
Should the quote be requested with the cart ID instead of quote ID?

Reply:
Since NQ REST API uses quote ID we have retained the same.
https://devdocs.magento.com/guides/v2.3/b2b/negotiable-manage.html#request-a-negotiable-quote

Reply:
Now I see where the confusion is coming from. We should not use existing REST APIs as a reference while building new GraphQL APIs.
Use already existing Magento GraphQL schema instead. 

There is no `quote_id` anywhere in our GraphQL schema, so it will be extremely confusing if we start using `quote_id` here while continue using `cart_id` in other places.

Reply:
We can still introduce `quote_id` to reference Negotiable quotes.

Reply:
Got it, So can we replace this with `cart_id` so that it will be uniform or can we introduce `quote_id` here. Thanks

Reply:
My understanding, to create a quote, we need to provide cart ID as an input, correct? If so, then it should be cart ID.

Reply:
Changed to `cart_id`.